aten/src/ATen/core/ivalue.cpp

```diff
@@ -199,6 +199,12 @@ void ivalue::Object::setAttr(const std::string& name, IValue v) {
   setSlot(slot, std::move(v));
 }
 
+void ivalue::Object::unsafeRemoveAttr(const std::string& name) {
+  const size_t slot = type_.type_->getAttributeSlot(name);
+  unsafeRemoveSlot(slot);
+  type_.type_->unsafeRemoveAttribute(name);
```

Comment:
This won't work in the case where there are multiple objects that have the same type. The first time, it will remove it from the type, then subsequent times will attempt to remove it and fail. This API should just remove a particular slot. `unsafeRemoveSlot`

Reply:
how about only allow remove attr if the type is used by one object? I don't feel it make much sense to remove the attribute of an object and don't remove the attribute in its underlying type.

Reply:
never mind, looks like use_count doesn't help me to find out how many objects uses this type.